cla_public/apps/checker/validators.py

```diff
@@ -73,21 +73,43 @@ def __call__(self, form, field):
         if len(field.data) < 1:
             message = self.message
             if message is None:
-                message = field.gettext("Must select at least one option.")
+                message = field.gettext("Select at least one option")
             raise ValidationError(message)
 
 
 class MoneyIntervalAmountRequired(object):
-    def __init__(self, message=None):
+    def __init__(self, message=None, freq_message=None, amount_message=None):
         self.message = message
+        self.freq_message = freq_message
+        self.amount_message = amount_message
+
+    def append_interval_text(self, field, interval):
+        if interval.data == "per_week":
```

Comment:
On reflection, I think this might be worth refactoring to avoid the multiple `if`s. Do you fancy pairing on it? Think it could be quite enjoyable.